atom/common/extensions/atom_extensions_client.cc

```diff
@@ -0,0 +1,149 @@
+#include "base/lazy_instance.h"
+#include "base/logging.h"
+#include "base/macros.h"
+#include "components/version_info/version_info.h"
+#include "content/public/common/user_agent.h"
+#include "extensions/common/core_extensions_api_provider.h"
+#include "extensions/common/extension_urls.h"
+#include "extensions/common/features/simple_feature.h"
+#include "extensions/common/permissions/permission_message_provider.h"
+#include "extensions/common/url_pattern_set.h"
+
+using extensions::ExtensionsClient;
+
+namespace atom {
+
+namespace {
+
+// TODO(jamescook): Refactor ChromePermissionsMessageProvider so we can share
+// code. For now, this implementation does nothing.
+class AtomPermissionMessageProvider
+    : public extensions::PermissionMessageProvider {
+ public:
+  AtomPermissionMessageProvider() {}
+  ~AtomPermissionMessageProvider() override {}
+
+  // PermissionMessageProvider implementation.
+  extensions::PermissionMessages GetPermissionMessages(
+      const extensions::PermissionIDSet& permissions) const override {
+    return extensions::PermissionMessages();
+  }
+
+  extensions::PermissionMessages GetPowerfulPermissionMessages(
+      const extensions::PermissionIDSet& permissions) const override {
+    return extensions::PermissionMessages();
+  }
+
+  bool IsPrivilegeIncrease(
+      const extensions::PermissionSet& granted_permissions,
+      const extensions::PermissionSet& requested_permissions,
+      extensions::Manifest::Type extension_type) const override {
+    // Ensure we implement this before shipping.
+    CHECK(false);
+    return false;
+  }
+
+  extensions::PermissionIDSet GetAllPermissionIDs(
+      const extensions::PermissionSet& permissions,
+      extensions::Manifest::Type extension_type) const override {
+    return extensions::PermissionIDSet();
+  }
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(AtomPermissionMessageProvider);
+};
+
+base::LazyInstance<AtomPermissionMessageProvider>::DestructorAtExit
+    g_permission_message_provider = LAZY_INSTANCE_INITIALIZER;
+
+}  // namespace
+
+AtomExtensionsClient::AtomExtensionsClient()
+    : webstore_base_url_(extension_urls::kChromeWebstoreBaseURL),
+      webstore_update_url_(extension_urls::kChromeWebstoreUpdateURL) {
+  AddAPIProvider(std::make_unique<extensions::CoreExtensionsAPIProvider>());
+  AddAPIProvider(std::make_unique<AtomExtensionsAPIProvider>());
+}
+
+AtomExtensionsClient::~AtomExtensionsClient() {}
+
+void AtomExtensionsClient::Initialize() {
+  // TODO(jamescook): Do we need to whitelist any extensions?
+}
+
+void AtomExtensionsClient::InitializeWebStoreUrls(
+    base::CommandLine* command_line) {}
+
+const extensions::PermissionMessageProvider&
+AtomExtensionsClient::GetPermissionMessageProvider() const {
+  NOTIMPLEMENTED();
+  return g_permission_message_provider.Get();
+}
+
+const std::string AtomExtensionsClient::GetProductName() {
+  // TODO(samuelmaddock):
+  return "app_shell";
```

Comment:
this should use atom again

Reply:
The issue here is that this in in `common` and therefore can run in either process.  We can't guarantee to have access to `atom::Browser::Get()->GetName()`

cc @deepak1556 @nornagon suggestions?

Reply:
where does this string end up? maybe just `Electron` is sufficient?